File: lab2/deserialize_json.py
# -*- coding: utf-8 -*-

import json

class DeserializeJson:
    # konstruktor
    def __init__(self, filename):
        print("let's deserialize something")
        with open(filename, encoding="utf8") as tempdata:
            self.data = json.load(tempdata)

    # # przykładowe statystyki
    # def somestats(self):
    #     example_stat = 0
    #     for dep in self.data:
    #         if dep['typ_JST'] == 'GM' and dep['Województwo'] == 'dolnośląskie':
    #             example_stat += 1
    #     print('liczba urzędów miejskich w województwie dolnośląskim: ' + str(example_stat))
    
    def somestats(self):
        wojewodztwa = {}  
        
        for dep in self.data:
            wojewodztwo = dep['Województwo']
            if wojewodztwo not in wojewodztwa:
                wojewodztwa[wojewodztwo] = 1  
            else:
                wojewodztwa[wojewodztwo] += 1  

        print("Liczba urzędów w poszczególnych województwach:")
        for wojewodztwo, liczba in wojewodztwa.items():
            print(f"{wojewodztwo}: {liczba}")

File: lab2z2_7/DeserializeJson.cs
using System;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

class DeserializeJson
{
    public List<Dictionary<string, object>> Data { get; private set; }

    public DeserializeJson(string filename)
    {
        Console.WriteLine("Deserializowanie JSON...");
        string jsonText = File.ReadAllText(filename);
        Data = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(jsonText);
    }

    public void ShowStats()
    {
        int count = 0;
        foreach (var item in Data)
        {
            if (item.ContainsKey("typ_JST") && item["typ_JST"].ToString() == "GM" &&
                item.ContainsKey("Województwo") && item["Województwo"].ToString() == "dolnośląskie")
            {
                count++;
            }
        }
        Console.WriteLine($"Liczba urzędów miejskich w województwie dolnośląskim: {count}");
    }
}
